pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ScalarTransformFunctionWrapper.java

```diff
@@ -113,6 +114,12 @@ public void init(List<TransformFunction> arguments, Map<String, DataSource> data
       }
     }
     _nonLiteralValues = new Object[_numNonLiteralArguments][];
+
+    // if FunctionInvoker contains an instance function. initialize it.
+    Object instance = _functionInvoker.getInstance();
+    if (instance != null && instance instanceof InitializableScalarFunction) {
```

Comment:
`instance instanceof InitializableScalarFunction` will fail (i.e. yield false) if the instance is null, so the null check is superfluous.

Reply:
did not know that. good catch